packages/components/organisms/f-form/src/services/formField.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable max-classes-per-file */
+class FormFieldClass {
+    constructor ({
+        name,
+        translations,
+        value,
+        maxLength = null
+    }) {
+        this.name = name;
+        this.value = value;
+        this.translations = translations;
+        this.validationMessages = this.translations.validationMessages || null;
+        this.maxLength = maxLength;
+        this.props = this.createProps();
+    }
+
+    createProps () {
+        const ariaLabel = this.createAriaLabel(this.value);
+
+        return {
+            name: this.kebabCase(),
+            'label-text': this.translations.label,
+            'input-type': this.inputType(),
+            ...this.maxLength && { 'max-length': this.maxLength },
+            ...ariaLabel && { 'aria-label': ariaLabel }
+        };
+    }
+
+    kebabCase () {
+        return this.name.replace(/([a-z0-9]|(?=[A-Z]))([A-Z1-9])/g, '$1-$2').toLowerCase();
```

Comment:
Out of interest, where did you get this regex from? Might be worth including the link in a comment